tests/functional/lti_certification/test_lti_certification_1_1.py

```diff
@@ -43,6 +54,15 @@ def assert_response_is_html(cls, response):
         assert response.headers["Content-Type"] == Any.string.matching("^text/html")
         assert response.html
 
+    def assert_redirected_to_tool_with_message(
+        self, response, lti_params, message=Any.string()
+    ):
+        expected_url = lti_params["launch_presentation_return_url"]
+        url = urlparse(response.headers["Location"])
+
+        assert url._replace(query=None).geturl() == expected_url
+        assert parse_qs(url.query) == Any.dict.containing({"lti_msg": [message]})
```

Comment:
Compare the URLs without query params first then check the dict contains the bit we care about. Other stuff is allowed too and will be preserved (see the LTIToolRedirect tests for that)